synapse/federation/transport/server.py

```diff
@@ -133,6 +134,9 @@ def authenticate_request(self, request, content):
         ):
             raise FederationDeniedError(origin)
 
+        if b"publicRooms" in request.path and self.deny_public_rooms:
```

Comment:
Not that I know of, because `BaseFederationServlet` and all other classes implementing it don't have direct access to the `hs`, therefore can't access the server's configuration. This is the simplest workaround I've found.

Reply:
You can pass in whatever you want to its constructor here: https://github.com/matrix-org/synapse/blob/develop/synapse/federation/transport/server.py#L1415

Reply:
Oh, I missed that, thanks.